Remove `--api-key` parameter

cdupuis opened the pull request:
As per #1, accepting `--api-key` is a security concern. Switching the `login` command to accept `--api-key-stdin`, `ATOMIST_API_KEY` env variable  or interactive login instead. 


fixes #1

cdupuis committed:
Remove --api-key parameter

fixes #1

Signed-off-by: Christian Dupuis <cd@atomist.com>

thaJeztah commented on main.go:
```diff
@@ -103,3 +106,44 @@ func main() {
+
+		apiKey = strings.TrimSuffix(string(contents), "\n")
+		apiKey = strings.TrimSuffix(apiKey, "\r")
+	} else if v, ok := os.LookupEnv("ATOMIST_API_KEY"); v != "" && ok {
```
So, it looks like the `ATOMIST_API_KEY` env-var is only read when calling the `docker base login` command, which then saves it to the `~/.docker/config.json` file; I wonder if users would expect the env-var to be usable as an _alternative_ to saving the token in the  config file, i.e.


```bash
# export ATOMIST_API_KEY=<my key>
...
...
docker base detect <some image>
```


Or can the API key _only_ be used in combination with a workspace?

cdupuis replied:
Yeah, the `API_KEY` requires a `workspace` to be set as well. You need both.

thaJeztah commented on main.go:
```diff
@@ -47,22 +53,26 @@ func main() {
+				if err != nil {
+					return err
+				}
+				if query.CheckAuth(workspace, apiKey) {
```
Not urgent, but I noticed that running `docker base login` without any flags set would produce an error as `workspace` was a required flag.

```bash
docker base login
required flag(s) "workspace" not set
```

From a usability perspective, it would be nice if in that case it would first prompt for the workspace, then ask for the API key, similar to `docker login`;

```bash
docker login
Login with your Docker ID to push and pull images from Docker Hub. If you don't have a Docker ID, head over to https://hub.docker.com to create one.
Username:
```

It's common to use flags for _optional_ arguments, and positional arguments as _required_ arguments, so if it's a _required_ parameter, then it could be `docker base login <workspace>`, but of course if it would be interactive, then `--workspace` would work for that.

cdupuis replied:
Ha, good point. I'll change that.

thaJeztah reviewed (approved):
I gave this a spin, and it looks to be working; left some comments/suggestions for possible follow-up improvements 😅

cdupuis commented:
> I gave this a spin, and it looks to be working; left some comments/suggestions for possible follow-up improvements 😅

Thanks @thaJeztah. I've addressed your 2nd comment and added interactive mode for the `workspace` as well.

thaJeztah commented:
>  I've addressed your 2nd comment and added interactive mode for the workspace as well.

Nice, thanks!
